test/e2e/network_segmentation_custom_l2udn.go

```diff
@@ -0,0 +1,138 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/ovn-org/ovn-kubernetes/test/e2e/feature"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	udnclientset "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/userdefinednetwork/v1/apis/clientset/versioned"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+var _ = Describe("Network Segmentation: Custom L2 UDN", feature.NetworkSegmentation, func() {
+	f := wrappedTestFramework("network-segmentation-custom-l2udn")
+	// disable automatic namespace creation, we need to add the required UDN label
+	f.SkipNamespaceCreation = true
+
+	var (
+		cs        clientset.Interface
+		udnClient udnclientset.Interface
+	)
+
+	BeforeEach(func() {
+
+		cs = f.ClientSet
+
+		var err error
+		udnClient, err = udnclientset.NewForConfig(f.ClientConfig())
+		Expect(err).NotTo(HaveOccurred())
+
+		namespace, err := f.CreateNamespace(context.TODO(), f.BaseName, map[string]string{
+			"e2e-framework":           f.BaseName,
+			RequiredUDNNamespaceLabel: "",
+		})
+		f.Namespace = namespace
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	DescribeTable("should respect network configuration",
+		func(netConfig *networkAttachmentConfigParams) {
+			By("creating the L2 network")
+
+			netConfig.namespace = f.Namespace.Name
+			udnManifest := generateUserDefinedNetworkManifest(netConfig)
+			cleanup, err := createManifest(netConfig.namespace, udnManifest)
+			Expect(err).NotTo(HaveOccurred())
+			DeferCleanup(cleanup)
+			Eventually(userDefinedNetworkReadyFunc(f.DynamicClient, netConfig.namespace, netConfig.name), 5*time.Second, time.Second).Should(Succeed())
+
+			By("creating a pod on the custom L2 network")
+			podConfig := *podConfig("gateway-test-pod")
+			podConfig.namespace = f.Namespace.Name
+			pod := runUDNPod(cs, f.Namespace.Name, podConfig, nil)
+
+			By("getting the created UDN object to validate against")
+			udn, err := udnClient.K8sV1().UserDefinedNetworks(netConfig.namespace).Get(context.Background(), netConfig.name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(udn.Spec.Layer2).NotTo(BeNil())
```

Comment:
nit: why are we validating the spec section of the UDN object we provision ? It feels like testing the test code.

Maybe I'm missing something.